sdks/go/pkg/beam/xlang.go

```diff
@@ -68,6 +68,45 @@ func CrossLanguagePayload(pl interface{}) []byte {
 	return bytes
 }
 
+// CrossLanguageWithClasspath is a low-level transform for executing cross-language transforms written in other
+// SDKs. Because this is low-level, it is recommended to use one of the higher-level IO-specific
+// wrappers where available. These can be found in the pkg/beam/io/xlang subdirectory.
+// CrossLanguage is useful for executing cross-language transforms which do not have any existing
+// IO wrappers. The functions beam.CrossLanguage() and beam.CrossLanguageWithClasspath() are similar except for
+// additional classpath dependencies that the later one accepts.
+//
+// Note: This should be specifically used when using an automated expansion service where you want to add additional
+// custom dependencies in the form of classpath to the expansion service.
+//
+// Usage requires an address for an expansion service accessible during pipeline construction, a
+// URN identifying the desired transform, an optional payload with configuration information,
+// input and output names, and classpath dependencies as an array of string. It outputs a map of named output PCollections.
+//
+// Example
+//
+// This example shows using CrossLanguageWithClasspath to execute the JDBC cross-language transform.
+//
+//    pl := beam.CrossLanguagePayload(JdbcConfigSchema)
+//    expansionAddr := xlang.UseAutomatedJavaExpansionService(gradleTargetOfExpansionService)
+//    readURN := ""
+//    classpath := []string{"org.postgresql:postgresql:42.3.3"}
+// 	  result := beam.CrossLanguageWithClasspath(s, readURN, pl, expansionAddr, nil, beam.UnnamedOutput(typex.New(outT)), classpath)
+func CrossLanguageWithClasspath(
+	s Scope,
+	urn string,
+	payload []byte,
+	expansionAddr string,
+	namedInputs map[string]PCollection,
+	namedOutputTypes map[string]FullType,
+	classpath []string,
+) map[string]PCollection {
```

Comment:
Instead of a new specific purpose method (in this case same as before but with a classpath parameter), lets try building it out as a new general purpose one. 

It should always have the scope, urn, payload, namedInputs, namedOutputs parameters, and a variadic xlangOption typed parameter.

This way we can have expansionAddr and class path become *optional* parameters, and enable some of the automatic start up stuff more directly, and less hacked in like it is presently.

No need to fully replace things under the hood for the others at this time, but I wouldn't say no.

Reply:
I guess building the expansion address handler can get rid of this. It is good to have though for users who want to use automated expansion service.